### ISSUE TITLE
TS type issues when using Lightswitch tables on fields Created/Modified (Datetimeoffset) and RowVersion(timestamp)

### ISSUE THREAD

**Issue by JohnBurgin** [volkanceylan/Serenity#1928]
We're looking to use Serenity to replace some systems written with MS Lightswitch. It seems like an ideal replacement but have ran into an early issue. If there is a need to re-run the T4 templates, it appears to create invalid data types. First time however is fine.

To recreate, I've made a new project to test

1. Run Sergen on a LS generated database
2. select table (in this case "Industries"), press generate
3. In VS, reload all
4. Transform T4
5. Run, and all is good

The Created and Modified datetimeoffests display are editiable strings and can only enter valid dates times. (later on, will look to make them read only, format as datatime and auto populate on save). The RowVersion is read only, and the value changes automatically on save.

If however I rerun the T4 transformation (due to changes in other tables), I get the following TS compile errors from the ServerTypings/Customers.IndustriesRow.ts
warning TS2694: Namespace 'System' has no exported member 'DateTimeOffset'
warning TS2694: Namespace 'System' has no exported member 'DateTimeOffset'
warning TS2694: Namespace 'System' has no exported member 'Byte'

First time through:

IntustriesRow.cs :
```
public class RowFields : RowFieldsBase
        {
            public Int32Field Id;
            public StringField Name;
            public StringField CreatedBy;
            public DateTimeOffsetField Created;
            public StringField ModifiedBy;
            public DateTimeOffsetField Modified;
            public ByteArrayField RowVersion;

```
Customers.IndustriesRow.ts :
  ```
  export interface IndustriesRow {
        Id?: number;
        Name?: string;
        CreatedBy?: string;
        Created?: any;
        ModifiedBy?: string;
        Modified?: any;
        RowVersion?: number[];
    }
```

6. Rebuild
7. Run T4
8. Rebuild - gives the TS errors

The generated types are now different and give the compile errors:
```
Customers.IndustriesRow.ts :
    export interface IndustriesRow {
        Id?: number;
        Name?: string;
        CreatedBy?: string;
        Created?: System.DateTimeOffset;
        ModifiedBy?: string;
        Modified?: System.DateTimeOffset;
        RowVersion?: System.Byte[];
    }
```

It's not a show stopper, as I can manually change these back to the original, but next time there is a need to run the T4 they break again. I'm sure the issue is not directly related to Lightswitch, but as all of the tables it creates have Created/Modifed Datetimeoffsets and RowVersions it is however causing some pain as we are working many tables.

To help recreate, here's the SQL for the table :

```
SET ANSI_NULLS OFF
GO

SET QUOTED_IDENTIFIER ON
GO

CREATE TABLE [dbo].[Industries](
	[Id] [int] IDENTITY(1,1) NOT NULL,
	[Name] [nvarchar](255) NOT NULL,
	[CreatedBy] [nvarchar](255) NULL,
	[Created] [datetimeoffset](7) NULL,
	[ModifiedBy] [nvarchar](255) NULL,
	[Modified] [datetimeoffset](7) NULL,
	[RowVersion] [timestamp] NOT NULL,
PRIMARY KEY CLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]

GO

ALTER TABLE [dbo].[Industries] ADD  DEFAULT ('') FOR [Name]
GO

```

**Comment by volkanceylan** [comment]
OK, thats probably a T4 generator type mapping issue, i don't use that field types so might not have seen this.

**Comment by volkanceylan** [comment]
This should be fixed in 2.9.5 for ServerTypings.tt, please check and let me know if any problems.
